pydatastructs/miscellaneous_data_structures/queue.py

```diff
@@ -0,0 +1,90 @@
+from pydatastructs.linear_data_structures import DynamicOneDimensionalArray
+
+__all__ = [
+    'Queue'
+]
+
+class Queue(object):
+    """Representation of queue data structure
+
+    Examples
+    ========
+
+    >>> from pydatastructs import Queue
+    >>> q = Queue()
+    >>> q.append(1)
+    >>> q.append(2)
+    >>> q.append(3)
+    >>> q.popleft()
+    >>> q.len()
+    1
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Queue_(abstract_data_type)
+    """
+
+    def __new__(cls, implementation='array', **kwargs):
+        if implementation == 'array':
+            return ArrayQueue(
+                kwargs.get('items', None),
+                kwargs.get('dtype', int))
+        raise NotImplementedError(
+                "%s hasn't been implemented yet."%(implementation))
+
+    def append(self, *args, **kwargs):
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    def popleft(self, *args, **kwargs):
+        raise NotImplementedError(
+            "This is an abstract method.")
+
+    @property
+    def is_empty(self):
+        return self.front == -1
+
+    @property
+    def is_full(self):
+        return ((self.rear + 1) % self.size == self.front)
+
+class ArrayQueue(Queue):
+
+    __slots__ = ['items', 'dtype']
+
+    def __new__(cls, items=None, dtype=int):
+        if items is None:
+            items = DynamicOneDimensionalArray(dtype, 0)
+        else:
+            items = DynamicOneDimensionalArray(dtype, items)
+        obj = object.__new__(cls)
+        obj.items, obj.dtype = \
+            items, items._dtype
+        return obj
+
+    def append(self, x):
+        if self.is_full:
+            raise ValueError("Queue is full")
+        elif (self.items.front == -1):
+            self.items.front = 0
```

Comment:
`self.items` is `DynamicOneDimensionalArray` and it doesn't have any attribute `front` and `rear`. Probably therefore the tests are failing.
One way is to make `front` and `rear` attributes of `ArrayQueue` so that you can replace `self.items.front` with `self.front` and `self.items.rear` with `self.rear`. That should fix the problem.

Reply:
@czgdp1807 thanks for the help